lahja/endpoint.py

```diff
@@ -129,6 +174,65 @@ def connect_no_wait(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> N
 
             self._process_item(item, config)
 
+    def _create_external_api(self, ipc_path: pathlib.Path) -> None:
+
+        receiver = EndpointConnector(self)
+
+        ConnectorManager = get_connector_manager()
+
+        ConnectorManager.register('get_connector', callable=lambda: receiver)  # type: ignore
+
+        manager = ConnectorManager(address=str(ipc_path))  # type: ignore
+        server = manager.get_server()   # type: ignore
+        threading.Thread(target=server.serve_forever, daemon=True).start()
+
+    def connect_to_endpoints_blocking(self, *endpoints: ConnectionConfig, timeout: int=30) -> None:
+        """
+        Connect to the given endpoints and block until the connection to every endpoint is
+        established. Raises a ``TimeoutError`` if connections do not become available within
+        ``timeout`` seconds (default 30 seconds).
+        """
+        for endpoint in endpoints:
+            wait_for_path_blocking(endpoint.path, timeout)
+            self._connect_to_endpoint(endpoint)
+
+    async def connect_to_endpoints(self, *endpoints: ConnectionConfig) -> None:
+        """
+        Asynchronously connect to the given endpoints and await until all connections are
```

Comment:
I don't think this counts as asynchronous, it blocks the caller until it's finished connecting!

Reply:
Well there's blocking and blocking. This blocks the caller in the same way `await asyncio.sleep(5)` "blocks" the caller but it does not block the thread and consequently not the event loop.

How about just:

>Connect to the given endpoints and await until all connections are established.

Reply:
Sounds good!